src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/MissingDeprecatedCheck.java

```diff
@@ -120,6 +123,13 @@ public final class MissingDeprecatedCheck extends AbstractJavadocCheck {
     public static final String MSG_KEY_JAVADOC_DUPLICATE_TAG =
             "javadoc.duplicateTag";
 
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY_UNCLOSED_HTML_TAG =
+            "javadoc.unclosedHtml";
```

Comment:
Will we still be able to find the message without this field? I expected us to only copy the message value into the `messages*.properties` files. Do all other checks that inherit from `AbstractJavadocCheck` "redeclare" this field?

I am not saying that either option (redeclare message key here or not) is good or bad, I am just curious if other checks do this or not - we should be consistent.

Reply:
All violations must be in a field for the XDoc/Meta scanner to pick them up. We can't pick up runtime violation keys.

If this field is declared in the abstract, then we don't need to put it here as it should be found while scanning the class file.

None of the other AsbtractJavadoc checks redeclare all the abstract messages as there are quite a few parse errors inherited.

Reply:
https://github.com/search?q=repo%3Acheckstyle%2Fcheckstyle%20javadoc.unclosedHtml&type=code

I don't see that other javadoc Checks are using this key of message. But we already agreed that it is beneficial for future to have all messages declared in exact check.

Reply:
As @romani mentioned, the key MSG_KEY_UNCLOSED_HTML_TAG is specific to MissingDeprecatedCheck.

Reply:
> I don't see that other javadoc Checks are using this key of message

Because it is used in abstract or other classes. Other javadoc checks don't need to reinvent the wheel.

As your code scan shows, message is declared in `JavadocDetailNodeParser` and used as violation in `AbstractJavadocCheck` and `JavadocStyleCheck`.

Reply:
I am ok to merge this PR as is.

Reply:
Ok, few things here:
1. As we can see in the diff of check messages in the xodcs file, we did not previously pick this message up since it was not actually declared as a message key (field) in AbstractJavadocCheck. This is a problem since we are likely not documenting this message in any javadoc check because of this.
2. We should be adding this field to AbstractJavadocCheck, and assigning to it the value from JavadocDetailNodeParser.MSG_KEY_UNCLOSED_HTML_TAG
3. We should update usage at https://github.com/checkstyle/checkstyle/blob/c9190ed2fbb919beafeb992bbce98912c6fa8717/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java#L322 accordingly.

Considering the above points, we should not be redeclaring this message in MissingDeprecatedCheck, we should fix the root cause in AbstractJavadocCheck.

```
➜  checkstyle git:(master) ag -Q -G '\.java' "javadoc.unclosedHtml"
src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocStyleCheck.java
172: * {@code javadoc.unclosedHtml}

src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java
79:    public static final String MSG_UNCLOSED_HTML_TAG = "javadoc.unclosedHtml";

src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/missingdeprecated/Example2.java
32:   * <p>  // violation, 'javadoc.unclosedHtml'
```

Reply:
We can fix the root cause in another issue/PR, but we should not redeclare the field in this class.

Reply:
>I don't see that other javadoc Checks are using this key of message. But we already agreed that it is beneficial for future to have all messages declared in exact check.

I do agree with this normally, but this seems like an exception since we are actually logging these violations in AbstractJavadocCheck.

Reply:
I approved issue, lets do it separately from this PR.

Reply:
@relentless-pursuit please remove message key declaration from check, but we can keep change in javadoc and xdoc since we actually use this message in examples. Other javadoc checks will get updated in scope of https://github.com/checkstyle/checkstyle/issues/13775

Reply:
@nrmancuso does it look good?

Reply:
@nrmancuso i am bit confused. if you are asking me to remove the message key declaration completely, one of the test case fails. 
[ERROR] Failures: 
[ERROR]   AllChecksTest.testAllCheckstyleMessages:572 property 'javadoc.unclosedHtml' isn't used by any check in package 'com.puppycrawl.tools.checkstyle.checks.annotation'
expected to contain: javadoc.unclosedHtml
but was            : [annotation.incorrect.style, annotation.parens.missing, annotation.trailing.comma.present, annotation.parens.present, annotation.trailing.comma.missing, suppressed.warning.not.allowed, annotation.location.alone, annotation.location, javadoc.parse.rule.error, javadoc.duplicateTag, annotation.missing.deprecated, javadoc.wrong.singleton.html.tag, javadoc.missed.html.close, tag.not.valid.on, annotation.missing.override, annotation.package.location, annotation.same.line]

And, i follow the same suite of JavadocStyleCheck in MissingDeprecatedCheck i.e. 
public static final String MSG_UNCLOSED_HTML = JavadocDetailNodeParser.MSG_UNCLOSED_HTML_TAG;

it complains of disallowed import. 
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/MissingDeprecatedCheck.java:24:1: Disallowed import - com.puppycrawl.tools.checkstyle.JavadocDetailNodeParser. [ImportControlMain]

Can you give me some insights on the same?

Reply:
@nrmancuso , did i understand your requested changes and executed in my recent push?

Reply:
@relentless-pursuit 

re: https://github.com/checkstyle/checkstyle/pull/13753#discussion_r1335222346

This is unfortunate, let's proceed with the field declaration as you had it before. I do not see any suppressions/exclusions for this test, so let's not start now.

Reply:
> @relentless-pursuit
> 
> re: [#13753 (comment)](https://github.com/checkstyle/checkstyle/pull/13753#discussion_r1335222346)
> 
> This is unfortunate, let's proceed with the field declaration as you had it before. I do not see any suppressions/exclusions for this test, so let's not start now.

done.